### PR TITLE
Implement nansum function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nansum, which computes the sum of elements along a specified axis while ignoring NaN values in the input tensor.
The operation follows NumPy semantics for axis and keepdims and is supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/backend/numpy/numpy.py [hunk 1]
```diff
@@ -960,6 +960,17 @@ def moveaxis(x, source, destination):
...
 
+def nansum(x, axis=None, keepdims=False):
+    axis = standardize_axis_for_numpy(axis)
+    dtype = standardize_dtype(x.dtype)
+
+    if dtype in ("bool", "int8", "int16"):
+        dtype = "int32"
+    elif dtype in ("uint8", "uint16"):
+        dtype = "uint32"
+    return np.nansum(x, axis=axis, keepdims=keepdims).astype(dtype)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/numpy/numpy.py:971]: np.nansum support axis=().

File: keras/src/backend/torch/numpy.py [hunk 2]
```diff
@@ -1272,6 +1272,19 @@ def moveaxis(x, source, destination):
...
+    if isinstance(x, (list, tuple)):
+        x = stack(x)
+    x = convert_to_tensor(x)
+    dtype = standardize_dtype(x.dtype)
+    if axis == () or axis == []:
+        return cast(torch.nan_to_num(x, nan=0), dtype)
+
+    if dtype in ("bool", "uint8", "int8", "int16"):
+        dtype = "int32"
+    return cast(torch.nansum(x, dim=axis, keepdim=keepdims), dtype)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/torch/numpy.py:1286]: 1. PyTorch does not support numeric operations on uint16, so it should not be included in upcasting logic for the Torch backend.
2. Applied.

File: keras/src/backend/tensorflow/numpy.py [hunk 3]
```diff
@@ -2125,6 +2125,26 @@ def moveaxis(x, source, destination):
...
+    if dtype in ("bool", "int8", "int16"):
+        dtype = "int32"
+    elif dtype in ("uint8", "uint16"):
+        dtype = "uint32"
+    x_clean = cast(x_clean, dtype)
+
+    if isinstance(x, tf.SparseTensor):
+        return tf.sparse.reduce_sum(
+            x_clean, axis=axis, keepdims=keepdims, output_is_sparse=True
+        )
```
Comment (hertschuh) [inline keras/src/backend/tensorflow/numpy.py]: If you really want to support `tf.SparseTensor`, you need to do something different line 2131. Also you would need to add unit tests. But honestly, I'm ok with you removing this block and not supporting `tf.SparseTensor`.

But for an example of how to do it, look at `take` and how do `sparse.sparse_with_values(x, fix_negative_indices(x))`.

https://github.com/keras-team/keras/blob/master/keras/src/backend/tensorflow/numpy.py#L2581-L2626

**Review by hertschuh — COMMENTED** [review]
Thanks for adding this!

File: keras/src/backend/tensorflow/numpy.py (hunk 3 above)
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py]: I’ve removed the block and am not attempting to support tf.SparseTensor at this time. If support is requested in the future, I will revisit this and address it then.
